src/Spice86.Core/Emulator/InterruptHandlers/Dos/Xms/ExtendedMemoryManager.cs

```diff
@@ -0,0 +1,490 @@
+﻿namespace Spice86.Core.Emulator.InterruptHandlers.Dos.Xms;
+
+using Serilog.Events;
+
+using Spice86.Core.Emulator.CPU;
+using Spice86.Core.Emulator.Extensions;
+using Spice86.Core.Emulator.InterruptHandlers;
+using Spice86.Core.Emulator.InterruptHandlers.Common.MemoryWriter;
+using Spice86.Core.Emulator.InterruptHandlers.Common.RoutineInstall;
+using Spice86.Core.Emulator.Memory;
+using Spice86.Shared.Emulator.Memory;
+using Spice86.Shared.Interfaces;
+using Spice86.Shared.Utils;
+
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+
+/// <summary>
+/// Provides DOS applications with XMS memory.
+/// </summary>
+public sealed class ExtendedMemoryManager : InterruptHandler, IMemoryDevice {
```

Comment:
At this point I would call it XmsManager and the EMS one EmsManager, Extended vs Expanded is confusing enough in the spec :)